docs(k6-studio): Add Assistant guided setup docs [NO MERGE] (#2281)
grafana/k6-docs: docs/sources/k6-studio/configure-test-with-assistant.md

weight: 360
---

# Configure a test with Grafana Assistant

Guided setup is a step-by-step wizard that uses [Grafana Assistant](https://grafana.com/docs/grafana-cloud/machine-learning/assistant/) to turn a recording into a configured HTTP test. At each step, the Assistant analyzes your recording and proposes configuration that you review and adjust: the hosts to include, correlation rules for dynamic values, parameterization rules for hard-coded values, and thresholds based on the response times in your recording. The final step summarizes the test plan and runs the test in [Grafana Cloud k6](https://grafana.com/docs/grafana-cloud/testing/k6/).

{{< admonition type="note" >}}

Guided setup is [experimental](https://grafana.com/docs/release-life-cycle/#experimental) and subject to change.

{{< /admonition >}}

Guided setup works with HTTP tests. To create a browser test from a recording, refer to [Record browser events](https://grafana.com/docs/k6-studio/record-browser-events/).

## Before you begin

To use guided setup, make sure you have:

> - **A Grafana Cloud account.** Guided setup uses Grafana Assistant, which requires a Grafana Cloud account. You can sign in, or create a free account, when the wizard prompts you.

Reviewer: ```suggestion
- **A [Grafana Cloud](https://grafana.com/products/cloud/) account.** Guided setup uses Grafana Assistant, which requires a Grafana Cloud account. You can sign in, or create a free account, when the wizard prompts you.
```

> - **A recording with requests.** Create a recording in k6 Studio, or open a HAR file.

Reviewer: ```suggestion
- **A recording with requests.** [Create a recording](https://grafana.com/docs/k6-studio/record-your-first-script/) in k6 Studio, or open a HAR file.
```

The proxy must be online for this step. If it's offline, you can skip the step and add correlation rules later.

### Parameterization

Hard-coded values, such as credentials and search terms, are baked into your recording. The Assistant finds them and creates parameterization rules that pull them into variables, so you can change the values in one place. Review the suggested parameters before continuing. After the wizard, you can edit the variable values in the Generator under **Test data** > **Variables**.

### Thresholds

Thresholds are the pass/fail criteria for your test. The Assistant proposes thresholds tuned to the response times it observes in your recording. Edit any value inline, or remove thresholds you don't need, before continuing.

### Run test

The last step summarizes what your test will do before you launch it:

- The number of requests and the hosts they run against.
- The thresholds that decide whether the test passes or fails.

> - The estimated cost of the run in virtual user hours (VUh), calculated by Grafana Cloud from your test options. If the configured load exceeds your project limits, a warning appears instead of the estimate.

Reviewer: ```suggestion
- The estimated cost of the run in virtual user hours (VUH), calculated by Grafana Cloud from your test options. If the configured load exceeds your project limits, a warning appears instead of the estimate.
```

- The load profile, with a timeline of the ramp-up, steady, and ramp-down stages.

> {{< figure src="/media/docs/k6-studio/screenshot-k6-studio-guided-setup-run-test-2.png" alt="Run test step in guided setup, summarizing requests, thresholds, estimated VU-hours, and the load stage timeline" >}}

Reviewer: ```suggestion
{{< figure src="/media/docs/k6-studio/screenshot-k6-studio-guided-setup-run-test-2.png" alt="Run test step in guided setup, summarizing requests, thresholds, estimated virtual user hours (VUH), and the load stage timeline" >}}
```

You can also expand **Edit load options** to adjust the load profile, review the **Configured by Assistant** summary of the earlier steps, and expand **View generated script** to inspect the k6 script.

> To launch the test, click **Save and run**. k6 Studio saves the test generator and opens the **Run in Grafana Cloud** dialog. When the run starts, the test results open in your browser. Refer to [Run a test in Grafana Cloud k6](https://grafana.com/docs/k6-studio/run-test-in-grafana-cloud/) for more details.

Reviewer: ```suggestion
To launch the test, click **Save and run**. k6 Studio saves the test generator and opens the **Run in Grafana Cloud** dialog box. When the run starts, the test results open in your browser. Refer to [Run a test in Grafana Cloud k6](https://grafana.com/docs/k6-studio/run-test-in-grafana-cloud/) for more details.
```
